### ISSUE TITLE
Snapping > Topological Editing is non-functional in Bucaresti

### ISSUE THREAD

**Issue by CMum** [qgis/QGIS#36429]
'Topological Editing' feature isn't functioning in Bucaresti, i.e. vertex of one shape may be moved in edit mode but vertex of joining shape in same layer does not automatically move with it.

Problem became evident when editing an SQLite layer. Same problem occurs when editing a .shp layer.

**Comment by gioman** [comment]
@CMum please add sample data. Do you have any evidence that with the very same data it was working as expected in a previous QGIS release?

**Comment by CMum** [comment]
Dear Mr. Manghi,

 

Thanks for your reply. 

 

Upon further investigation of the reported problem, I realize that I have topological editing capability on some but not all layers. Unfortunately, I can’t find a setting that would be toggling functionality between layers. The only difference between functioning vs. non-functioning layers that I have noticed is whether or not the layer is my creation.  For example, a layer with expected topological functionality is an SQLite layer of my own creation, including new, i.e. original polygon data of my own creation. On the other hand, ‘public’ shapefile layers, i.e. shapefiles that I’ve imported from an external source, don’t have expected functionality. This includes an SQLite layer of my creation to which I exported one of the ‘public’ shapefiles. 

 

FYI: I’m working with small parcels in multipolygon layers. 

 

Bottom line, topological editing is functional on some but not all layers. I am unable to determine why it is not functional on all layers.

 

Thanks for your assistance.

 

Sincerely,

 

Christopher Mumford

Plymouth NH USA

**Comment by gioman** [comment]
> On the other hand, ‘public’ shapefile layers, i.e. shapefiles that I’ve imported from an external source, don’t have expected functionality. This includes an SQLite layer of my creation to which I exported one of the ‘public’ shapefiles.

@CMum this means that is likely that in in this vectors you have vertexes that seems to be in the very same location/coordinate, but are not.

Please attach/link a sample of such vectors.

**Comment by CMum** [comment]
I also considered this possibility. The snapping feature is functional on all layers, and so I “resnapped” some vertices on one of the problem layers, saved, and then tried the topological feature. Unfortunately, it didn’t work as expected. 

 

Next, I selected a few shapes in the ‘problem’ layer I most want to have topological editing. The original layer is an SQLite layer and I exported to another SQLite layer. Upon testing the ‘new’ layer, it now has the expected Topological functionality. 

 

As a workaround, I retraced steps I took above, i.e. created a new layer from the non-functioning layer, which indeed has the desired functionality. 

 

Would you like to receive a copy of the original non-functional layer? It should be relatively small when zipped. 

 

Chris M.

**Comment by gioman** [comment]
>  Would you like to receive a copy of the original non-functional layer? It should be relatively small when zipped.

@CMum attach/link it here, do not send it to me directly.

**Comment by CMum** [comment]
See attachment. 

FYI:: This was a 'public' shapefile that I subsequently exported / saved as an SQLite layer. At present, some of the original shapes have been edited. I have also added some entirely new shapes.
[FSA_2019.zip](https://github.com/qgis/QGIS/files/4628929/FSA_2019.zip)

**Comment by gioman** [comment]
@CMum why you have closed the ticket?

**Comment by CMum** [comment]
As one other comment, I just tested the original shapefile, and discovered that topological editing works as expected. The layer I sent is the one that non-functional.

**Comment by gioman** [comment]
> FYI:: This was a 'public' shapefile that I subsequently exported / saved as an SQLite layer. At present, some of the original shapes have been edited. I have also added some entirely new shapes.

@CMum can you be more specific? we just need 1 example of a vector where the topological editing does not work as expected.

**Comment by gioman** [comment]
> The layer I sent is the one that non-functional.

@CMum thanks. Why closing?

**Comment by CMum** [comment]
Sorry, that was a mistake. My first time using GitHub. Is it still closed? What should I do to recover?

**Comment by CMum** [comment]
Again, I didn’t intend to close. Hopefully the thread is still open.

**Comment by gioman** [comment]
> FSA_2019.zip

@CMum no issue here  with topological editing on this layer.

**Comment by CMum** [comment]
Wow.  I’m not sure what is happening or not happening. 

 

In any event, thanks for the time you’ve given me. I declare it a non-issue and will close the thread.
